projects/public-search/src/app/patron-profile/patron-profile-personal-editor/patron-profile-personal-editor.component.ts

```diff
@@ -0,0 +1,203 @@
+                // Add a class on each field
+                field.templateOptions.itemCssClass = (fkey in this.cssConfig)
+                  ? this.cssConfig[fkey]
+                  : this.cssConfig.default;
+                // Field deactivation
+                if (field.key !== undefined && disabledFields.includes(fkey)) {
+                  field.templateOptions.readonly = true;
+                }
+                // Hide password field
+                if (field.key === 'password') {
+                  field.templateOptions.readonly = true;
+                  field.hide = true;
+                }
+                // Add description if exists
+                if (Object.keys(this.fieldDescription).includes(fkey)) {
+                  field.templateOptions.description = this.fieldDescription[fkey];
+                }
+                // Add validator on field
+                switch (fkey) {
+                  case 'email':
+                    if (field.asyncValidators == null) {
+                      field.asyncValidators = {};
+                    }
+                    field.asyncValidators.uniqueEmail = this._getUniqueValidator('email');
+                    break;
+                  case 'username':
+                    if (field.asyncValidators == null) {
+                      field.asyncValidators = {};
+                    }
+                    field.asyncValidators.uniqueUsername = this._getUniqueValidator('username');
+                }
+                // remove Message suffix to the message validation key
+                // (required for backend  translations)
+                if (field.validation) {
+                  Object.keys(field.validation.messages).map(msg => {
+                    if (msg.endsWith('Message')) {
+                      const val = field.validation.messages[msg];
+                      delete (field.validation.messages[msg]);
+                      const newMsg = msg.replace(/Message$/, '');
+                      field.validation.messages[newMsg] = val;
+                    }
+                  });
+                }
+                return field;
+              }
+            })
+          ];
+        }
+      })
+    );
+
+    const userQuery = this._recordService.getRecord('users', this.user.id);
+
+    forkJoin([schemaForm, userQuery]).subscribe(([model, user]: [any, any]) =>
+      this.model = user.metadata);
+  }
+
+  submit() {
+    this.form.updateValueAndValidity();
+    if (this.form.valid === false) {
+      this._toastrService.error(
+        this._translateService.instant('The form contains errors.')
+      );
+      return;
+    }
+
+    const data = removeEmptyValues(this.form.value);
+
+    // Update user record and reload logged user
+    this._recordService.update('users', this.user.id, data).subscribe(() => {
+      this._userService.load().subscribe((user: any) => {
+        this.userEvent.emit(user);
+        this.cancelEvent.emit(true);
```

Comment:
Why a cancel event here?

Reply:
It's just to show the personal data screen